sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/GatewayConnectionConfig.java

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import java.net.InetSocketAddress;
+import java.time.Duration;
+
+import static com.azure.cosmos.implementation.ConnectionPolicy.DEFAULT_IDLE_CONNECTION_TIMEOUT;
```

Comment:
you're consuming implementation package here and the class is public facing...

Reply:
just to be clear, this can be done after GA :-)

Reply:
why is this a problem?

Reply:
@milismsft 's point is - Since we are using default values from implementation details - changing these default values in implementation will have a direct impact on customer's app, which can be considered breaking change in some way. 
I think its a valid point, I will think what can be done for this.

Reply:
The `RntbdTransportClient` will treat the `DirectConnectionConfig` as definitive. Regardless of where we put the default values, changes to them will affect behavior, and changes to behavior must be rationalized and documented. This sounds like a good reason to move the public-facing defaults to the `ConnectionConfig` classes. It will make it easier for us to understand the impact of a change. The transport clients could then pull defaults from their respective public config classes; not the other way around.

Reply:
Done.

Reply:
Updated constants to GatewayConnectionConfig.java